python/ray/_private/worker.py

```diff
@@ -852,13 +877,25 @@ def put_object(
         # reference will be created. If another reference is created and
         # removed before this one, it will corrupt the state in the
         # reference counter.
-        return self.core_worker.put_object(
+        ret = self.core_worker.put_object(
             serialized_value,
             pin_object=pin_object,
             owner_address=owner_address,
             inline_small_object=True,
             _is_experimental_channel=_is_experimental_channel,
+            tensor_transport_val=tensor_transport.value,
         )
+        if tensors:
+            src_actor = ray.get_runtime_context().current_actor
+            self.get_serialization_context().store_gpu_objects(ret.hex(), tensors)
```

Comment:
I think it is clearer to just directly call `gpu_object_store.add_object` instead of going through the serialization context.